### PR TITLE
Add clang-tidy, include-what-you-use, and msvc-analyzer github actions

### PR DESCRIPTION
**Pull request by jonesmz**
These code scanners can provide a lot of great automated bug finding capabilities.

### CODE REVIEW

**Comment by eakraly** [comment]
Thanks @jonesmz !
Could you please review issues with the new actions?

**Comment by jonesmz** [comment]
> Thanks @jonesmz ! Could you please review issues with the new actions?

Hi

Unfortunately, this failure appears to be happening in the main branch as well.

https://github.com/coturn/coturn/actions/runs/7488070087/job/20381752064

The failure appears to be something wrong with vcpkg ?

**Comment by eakraly** [comment]
> > Thanks @jonesmz ! Could you please review issues with the new actions?
> 
> Hi
> 
> Unfortunately, this failure appears to be happening in the main branch as well.
> 
> [coturn/coturn/actions/runs/7488070087/job/20381752064](https://github.com/coturn/coturn/actions/runs/7488070087/job/20381752064)
> 
> The failure appears to be something wrong with vcpkg ?

@jonesmz can you pls rebase on top of master and try again?

**Comment by jonesmz** [comment]
> > > Thanks @jonesmz ! Could you please review issues with the new actions?
> > 
> > 
> > Hi
> > Unfortunately, this failure appears to be happening in the main branch as well.
> > [coturn/coturn/actions/runs/7488070087/job/20381752064](https://github.com/coturn/coturn/actions/runs/7488070087/job/20381752064)
> > The failure appears to be something wrong with vcpkg ?
> 
> @jonesmz can you pls rebase on top of master and try again?

I rebased and copied the changes you made to the windows build to also apply to the msvc-analyzer.yml file.

**Comment by jonesmz** [comment]
Note: @eakraly while the clang-tidy workflow doesn't error out on new issues detected, it does have several reports of potential nullptr dereferences that would be worth looking into. I only reported just one in an issue.

File: .clang-tidy [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+Checks: 'clang-diagnostic-*,
```
Comment (jonesmz) [inline .clang-tidy:2]: Note also that i tried to construct an analysis list that balanced "too annoying" with "not helpful enough".

It's possible to disable some of the more frequent analyzer reports by making further modifications to this file.

**Pull request merged**